samples/Samples.Mvc5/Controllers/HomeController.cs

```diff
@@ -131,6 +133,34 @@ public ActionResult About()
         /// <returns>The <see cref="ActionResult"/>.</returns>
         public ActionResult ResultsAuthorization() => View();
 
+        public async Task<ActionResult> CallAsync()
+        {
+            var profiler = MiniProfiler.Current;
+            var service = new FooService();
+            using (profiler.Step("action"))
+            {
+                var head = profiler.Head;
+                var tasks = new List<Task>();
+                for (int i = 0; i < 10; i++)
+                {
+                    tasks.Add(service.FooAsync(head));
+                }
+                await Task.WhenAll(tasks);
+                return Content("All good");
+            }
+        }
+
+        class FooService
+        {
+            public async Task FooAsync(Timing head)
+            {
+                using (new Timing(MiniProfiler.Current, head, "foo async", minSaveMs: 100000))
```

Comment:
Also high minSaveMs is important since it triggers the children to be removed from the parent node, if their timing exceeds minSaveMs or minSaveMs is not set they would not be removed and the issue will not get triggered - this is another artificial piece about this test.